index.js

```diff
@@ -81,42 +81,60 @@ async function authenticate (_retryCount = 1) {
   }
 }
 
-async function _retryAuth (_retryCount) {
-  if (_retryCount <= 3) {
-    logger.warning(`Authentication retry #${_retryCount} due to empty response from Sierra API`)
-    await delay(1000 * Math.pow(2, _retryCount - 1))
-    await authenticate(_retryCount + 1)
-  } else {
-    const error = new RetryError('Authentication')
-    logger.error(error.message)
-    throw error
-  }
-}
-
 async function get (path, _retryCount = 1) {
   try {
-    await authenticate()
-    const response = await axios.get(credsBase + path, {
-      timeout: 120 * 1000, headers: { Authorization: `Bearer ${accessToken}` }
-    })
+    const response = await _doHttpRequest('get', path)
     if (response.data === '' && response.status < 300 && response.status >= 200) {
       await _retryGet(path, _retryCount)
     } else {
       return response.data
     }
   } catch (error) {
-    if (error.response) {
-      if (error.response.status === 401) {
-        await _reauthenticate()
-        return get(path)
-      }
-    }
+    return await _handleErrors(error, deleteRequest, path)
+  }
+}
+
+async function post (path, data) {
+  try {
+    const response = await _doHttpRequest('post', path, data)
+    return response.data
+  } catch (error) {
+    return await _handleErrors(error, deleteRequest, path)
+  }
+}
+
+async function put (path, data) {
+  try {
+    const response = await _doHttpRequest('put', path, data)
+    return response.data
+  } catch (error) {
+    return await _handleErrors(error, deleteRequest, path)
```

Comment:
This too should pass `put`, `path`, and `data`?

Reply:
yes good catch!